--- index.js.orig	2011-04-22 19:23:57.000000000 +0400
+++ index.js	2011-04-22 19:27:37.000000000 +0400
@@ -473,7 +473,11 @@
         return;
     }
 
-    if (command === "subscribe" || command === "psubscribe" || command === "unsubscribe" || command === "punsubscribe") {
+    if (command === 'mget' && args.length === 0) {
+        // we skip mget([])
+        if (callback) callback(null, []);
+        return;
+    } else if (command === "subscribe" || command === "psubscribe" || command === "unsubscribe" || command === "punsubscribe") {
         if (this.subscriptions === false && exports.debug_mode) {
             console.log("Entering pub/sub mode from " + command);
         }
